compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -259,6 +260,44 @@ impl<'a, 'tcx> CfgSimplifier<'a, 'tcx> {
     }
 }
 
+pub fn remove_duplicate_unreachable_blocks<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+    struct OptApplier<'tcx> {
+        tcx: TyCtxt<'tcx>,
+        first: BasicBlock,
+        duplicates: FxHashSet<BasicBlock>,
+    }
+
+    impl<'tcx> MutVisitor<'tcx> for OptApplier<'tcx> {
+        fn tcx(&self) -> TyCtxt<'tcx> {
+            self.tcx
+        }
+
+        fn visit_terminator(&mut self, terminator: &mut Terminator<'tcx>, location: Location) {
+            for target in terminator.successors_mut() {
+                if self.duplicates.contains(target) {
+                    *target = self.first;
+                }
+            }
+
+            self.super_terminator(terminator, location);
+        }
+    }
+
+    let unreachable_blocks = body
+        .basic_blocks
+        .iter_enumerated()
+        .filter_map(|(block, bb)| {
+            if bb.terminator.is_some() && bb.is_empty_unreachable() { Some(block) } else { None }
```

Comment:
```suggestion
            bb.is_empty_unreachable().then_some(block)
```
`bb.terminator` must always be `Some`.

Reply:
This change trips the assertion: https://github.com/rust-lang/rust/actions/runs/4264669115/jobs/7423048035#step:26:1216

Reply:
Oh. `CfgSimplifier` leaves behind some unreachable basic blocks without a terminator. `remove_dead_blocks` cleans up after it, so it's not really an issue. Could you just add a comment then? Checking the terminator to be `Some` is not standard practice.